src/Viewer.tsx

```diff
@@ -355,6 +358,13 @@ function Viewer(): ReactElement {
   );
 
   // DATASET LOADING ///////////////////////////////////////////////////////
+
+  const handleProgressUpdate = useCallback((complete: number, total: number): void => {
+    startTransition(() => {
+      setDatasetLoadProgress(Math.round((complete / total) * 100));
+    });
```

Comment:
Many updates are expected to occur, so use a transition to mark the render as interruptible and non-critical.